build_tools/docker/context/install_basics.sh

```diff
@@ -12,4 +12,5 @@ set -euo pipefail
 # (https://docs.docker.com/develop/develop-images/multistage-build/) but it
 # turns out that Dockerfile is a thoroughly non-composable awful format and that
 # doesn't actually work that well. These deps are pretty small.
-apt-get update && apt-get install -y git unzip wget curl gnupg2 lsb-release
+apt-get update && apt-get install -y git unzip wget curl gnupg2 lsb-release sudo
+echo "ALL ALL=NOPASSWD: ALL" > "/etc/sudoers.d/iree"
```

Comment:
This seems fine to me, but out of curiosity, what do you want to install during job execution that needs to be under a docker image? (I'm aiming to remove more uses of Docker, so I want to limit new dependencies on it)

Reply:
I want specifically `libopenmpi-dev`. What would we use instead of docker?

Reply:
Ah, I see. Installing the Python `mpi4py` package still needs libopenmpi-dev: https://stackoverflow.com/questions/28440834/error-when-installing-mpi4py (https://github.com/mpi4py/setup-mpi even uses apt get too).

My overall goals are for build machines and jobs to be separate from test machines and jobs.
* The compiler is built on host platforms (Linux, Windows, macOS, arm/x86, etc.), using Docker for toolchains (CMake, Bazel, Ninja, MSVC, clang, etc.) as needed
* Tests (think "user workflows") are run on target machines using readily available packages, drivers, and other deps. For Python that means pip installing some utilities. For GPU execution that means installing drivers - either from the test runner machine itself, from a standard dockerfile (maybe https://hub.docker.com/r/nvidia/cuda ?), or from a custom dockerfile.